src/openapi/streaming/subscription.js

```diff
@@ -218,13 +228,20 @@ function onSubscribeError(referenceId, response) {
         return;
     }
 
-    this.currentState = STATE_UNSUBSCRIBED;
-    log.error(LOG_AREA, 'An error occurred subscribing', { response, url: this.url });
+    if (response && response.status === '504' && this.retryCounter++ < MAX_RETRIES) {
+        log.error(LOG_AREA, 'Received gateway timeout when subscribing. Retrying.', { response, url: this.url });
 
-    // if we are unsubscribed, do not fire the error handler
-    if (this.queue.peek() !== ACTION_UNSUBSCRIBE) {
-        if (this.onError) {
-            this.onError(response);
+        this.currentState = STATE_SUBSCRIBED; // set to subscribed so that unsubscribe request isn't ignored
+        retrySubscription.call(this);

```

Comment:
It looks like we will run two actions in such case. First, retrySubscription will run unsubscribe immediately. And at the end of this function we are calling:  
```
onReadyToPerformNextAction.call(this);
```
Which will run next action from a queue. I don't think it's quite right.